lib/features/feed/cubit/share_feed_post_cubit.dart

```diff
@@ -1,30 +1,104 @@
         final files = await Misc.downloadFiles([feed.image!]);
-        await Share.shareXFiles(
-          files,
-          text: '${feed.body}\n\nhttps://fluttercon.dev',
-        );
-      } else {
-        await Share.share('${feed.body}\n\nhttps://fluttercon.dev');
+        if (files.isNotEmpty) {
+          filePath = files.first.path;
+        }
       }
+
+      final installedApps = await appinioSocialShare.getInstalledApps();
+
+      switch (platform) {
+        case Platform.telegram:
+          if (installedApps['telegram'] ?? false) {
+            await _shareToTelegram(message, filePath);
+          } else {
+            emit(const ShareFeedPostState.error('Telegram is not installed.'));
+          }
+        case Platform.whatsapp:
+          if (installedApps['whatsapp'] ?? false) {
+            await _shareToWhatsApp(message, filePath);
+          } else {
+            emit(const ShareFeedPostState.error('WhatsApp is not installed.'));
+          }
+        case Platform.twitter:
+          if (installedApps['twitter'] ?? false) {
+            await _shareToTwitter(message, filePath);
+          } else {
+            emit(const ShareFeedPostState.error('Twitter is not installed.'));
+          }
+        case Platform.facebook:
+          if (installedApps['facebook'] ?? false) {
+            if (filePath == null) {
+              emit(
+                const ShareFeedPostState.error(
+                  'Failed to share post. Kindly share a post with an image.',
+                ),
+              );
+            } else {
+              await _shareToFacebook(message, filePath);
+            }
+          } else {
+            emit(const ShareFeedPostState.error('Facebook is not installed.'));
+          }
+      }
+
       emit(const ShareFeedPostState.loaded());
     } catch (e) {
-      emit(ShareFeedPostState.error(e.toString()));
+      emit(ShareFeedPostState.error('Failed to share post: $e'));
+    }
+  }
+
+  Future<void> _shareToTelegram(String message, String? filePath) async {
+    try {
+      await appinioSocialShare.android.shareToTelegram(message, filePath);
+    } catch (e) {
+      throw Exception('Error sharing to Telegram: $e');
+    }
+  }
+
+  Future<void> _shareToWhatsApp(String message, String? filePath) async {
+    try {
+      await appinioSocialShare.android.shareToWhatsapp(message, filePath);
+    } catch (e) {
+      throw Exception('Error sharing to WhatsApp: $e');
+    }
+  }
+
+  Future<void> _shareToFacebook(String message, String filePath) async {
+    try {
+      await appinioSocialShare.android.shareToFacebook(message, [filePath]);
+    } catch (e) {
+      throw Exception('Error sharing to Facebook: $e');
+    }
+  }
+
+  Future<void> _shareToTwitter(String message, String? filePath) async {
+    try {
+      await appinioSocialShare.android.shareToTwitter(message, filePath);
+    } catch (e) {
+      throw Exception('Error sharing to Twitter: $e');
     }
   }
```

Comment:
The best place to have these individual functions would be a repository. Think of it this way. 

Repos - Direct interaction with the platform
Cubits - Updating the UI with the state
Widget - Results of the operations

The sharing function may need to be used in another instance, `lib/features/sessions/ui/session_details/session_details.dart` so best to have that independent of which screen it's used in. Please make a repo file for the functions like so:
- In `lib/common/repository`, make `share_repository.dart`
- Inside, define the class `class ShareRepository {` and ensure to annotate it with `@singleton` as seen in `lib/common/repository/api_repository.dart:10`. Run code gen `gen:` command in the `Makefile`. This ensures that the class is available for injection
- Then you can inject it into `ShareFeedPostCubit`. You can see a sample of the injection of other cubits in  the `providers` array `lib/bootstrap.dart:54`
- Then inside the cubit, you have a much neater file. 
- In the `api_repository.dart` file, you can observe how the try...catch happens. If you want to throw a particular error, throw the `Failure (lib/common/data/models/failure.dart)` then you can ensure to catch it in the cubit. Sample catch: `lib/features/about/cubit/fetch_individual_organisers_cubit.dart:32-34`

Reply:
I will work on this first thing in the morning.